### PR TITLE
PR: Restore sorting of types and natural sorting

### LINKED ISSUE #13733: variable type and size are out of order
## Problem Description

Variable explorer problem, variable type and size are out of order

## Test Code
```python
# -*- coding: utf-8 -*-
import numpy as np
import pandas as pd
class IndDistIdent(object):
    def __init__(self):
        self.algName = {'algName':'algName'}
        self.keysZhCN = {'keysZhCN':
                             {
                                 'returnName':'returnName'
                             }
                        }
    def main(self,data):
        desSta = self.__desSta(data)
        ret = {**desSta,**self.algName,**self.keysZhCN}
        print(type(ret.get('algName')),len(ret.get('algName')))
        print(type(ret.get('DesSta')),len(ret.get('DesSta')))

        return ret

    def __desSta(self,data):
        n = len(data)
        mean = data.mean()
        sigma = data.std()
        med = data.median()
        minV = data.min()
        maxV = data.max()
        skew = data.skew()
        kurt = data.kurt()
        values = [n,mean,sigma,med,minV,maxV,skew,kurt]
        index = ['N','Mean','Sigma','Median','Min','Max','Skew','Kurt']
        ret = pd.Series(values,index=index,name='DesSta')
        return {'DesSta':ret}

if __name__ == '__main__':
    data = pd.Series(np.random.normal(loc=0,scale=1,size=100))
    idi = IndDistIdent()
    ret = idi.main(data)
```

### What steps reproduce the problem?
 
spyder4 1 5变量查看bug

The type and size of the algName and desSta variables in ret are out of order

## Versions

* Spyder version: 4.1.5
* Python version: 3.7.6
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: win10

### PR DESCRIPTION
**Pull request by skjerns**
related: https://github.com/spyder-ide/spyder-kernels/pull/246

I've re-implemented the natural sorting, now it should also work when sorting for types, etc.

### Issue(s) Resolved

Fixes #13733 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
skjerns

### CODE REVIEW

**Comment by skjerns** [comment]
I don't quite understand why the merging is blocked, can someone enlighten me? according to https://github.com/spyder-ide/spyder/pull/13791/files all should be fine

**Comment by dalthviz** [comment]
Hi @skjerns I think the base was set to `master` instead of `4.x` Thanks for the help with this one :+1:

File: spyder/plugins/variableexplorer/widgets/collectionseditor.py [hunk 1]
```diff
@@ -1817,6 +1825,6 @@ def remote_editor_test():
     app.exec_()
 
 
-if __name__ == "__main__":
-    editor_test()
-    remote_editor_test()
+# if __name__ == "__main__":
```
Comment (dalthviz) [inline spyder/plugins/variableexplorer/widgets/collectionseditor.py:1828]: Should this be commented out? If needed, could you explain why please?

File: spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py [hunk 2]
```diff
@@ -845,5 +846,59 @@ def test_dicts_natural_sorting(qtbot):
         'GUI sorting fail'
 
 
+
+def test_dicts_natural_sorting_mixed_types(qtbot):
```
Comment (dalthviz) [inline spyder/plugins/variableexplorer/widgets/tests/test_collectioneditor.py:850]: Awesome :+1:

**Review by dalthviz — COMMENTED** [review]
Thanks @skjerns for your help! I left a comment and seems like some code style issues were raised. Besides that this one LGTM :+1:

File: spyder/plugins/variableexplorer/widgets/collectionseditor.py (hunk 1 above)
Comment (skjerns) [reply spyder/plugins/variableexplorer/widgets/collectionseditor.py:1828]: aah, I forgot to comment it back in. I can't run the file (and put the declarations in the workspace)

**Comment by skjerns** [comment]
> Hi @skjerns I think the base was set to `master` instead of `4.x` Thanks for the help with this one 👍

Ah! That makes sense.



> > [Line 263:9](https://github.com/spyder-ide/spyder/blob/0543c3033be41742e5bfe73ca6bf576f7842f646/external-deps/spyder-kernels/spyder_kernels/utils/nsview.py#L263): [E731](https://duckduckgo.com/?q=pep8%20E731) do not assign a lambda expression, use a def
> > [Line 265:9](https://github.com/spyder-ide/spyder/blob/0543c3033be41742e5bfe73ca6bf576f7842f646/external-deps/spyder-kernels/spyder_kernels/utils/nsview.py#L265): [E731](https://duckduckgo.com/?q=pep8%20E731) do not assign a lambda expression, use a def
> 

I would like to keep it as a `lambda`. I think a lambda is far more comprehensible in this case. What do you think, can we make an exception?

**Comment by dalthviz** [comment]
Thanks for the changes @skjerns I think that using the lambda is fine :+1: 

Now I will merge the spyder-kernels PR and then we will need to update the git-subrepo of the kernel here (that will also re-trigger the CI and hopefully the flaky test in mac will pass without issues). After that this should be ready to be merged :)

**Comment by dalthviz** [comment]
@skjerns I merged spyder-ide/spyder-kernels#246 now the missing thing is to update the git subrepo of the kernel to have the latest version in `1.x`. Let us know if you want to do that or if you want me to take care of that. Thanks again for working on this one!

**Comment by skjerns** [comment]
@dalthviz can you do that? I had quite some trouble setting up the subrepo under Windows

**Review by dalthviz — APPROVED** [review]
Thanks @skjerns LGTM :+1:

**Pull request merged**